### PR TITLE
Use 'diff_base' for pycodestyle check, if set.

### PR DESCRIPTION
**Pull request by tseaver**
Speeds it up, plus reduces spew when testing locally.

### CODE REVIEW

File: scripts/pycodestyle_on_repo.py [hunk 1]
```diff
@@ -23,14 +23,17 @@
...
+
 
 def main():
     """Run pycodestyle on all Python files in the repository."""
     git_root = subprocess.check_output(
         ['git', 'rev-parse', '--show-toplevel']).strip()
     os.chdir(git_root)
-    python_files = subprocess.check_output(['git', 'ls-files', '*py'])
-    python_files = python_files.strip().split()
+    candidates, _ = get_files_for_linting()
```
Comment (daspecster) [inline scripts/pycodestyle_on_repo.py]: Shouldn't `get_files_for_linting()` only return `.py` files?

Comment (daspecster) [inline scripts/pycodestyle_on_repo.py]: Actually there's a `get_python_files()` method that might be helpful here.
https://github.com/GoogleCloudPlatform/google-cloud-python/blob/72291ea6105e8a53761e03abd15bda5bd8c53ee5/scripts/run_pylint.py#L196

**Comment by dhermes** [comment]
@tseaver I recently introduced `script_utils` for shared code like this. Care to move?

File: scripts/pycodestyle_on_repo.py (hunk 1 above)
Comment (tseaver) [inline scripts/pycodestyle_on_repo.py]: `get_python_files` chunks the set into "library" vs. "tests" -- it is easier here just to search them myself.

File: scripts/pycodestyle_on_repo.py [hunk 2]
```diff
@@ -23,14 +23,17 @@
 import subprocess
 import sys
 
+from script_utils import get_files_for_linting
```
Comment (dhermes) [inline scripts/pycodestyle_on_repo.py]: @tseaver Shouldn't we rename it now? How about `get_affected_files`?

File: scripts/script_utils.py [hunk 3]
```diff
@@ -159,3 +161,55 @@ def get_changed_packages(blob_name1, blob_name2, package_list):
...
+    To allow faster local ``tox`` runs, the local remote and local branch
+    environment variables can be set to specify a remote branch to diff
+    against.
+
+    :type allow_limited: bool
+    :param allow_limited: Boolean indicating if a reduced set of files can
+                          be used.
+
+    :rtype: pair
+    :returns: Tuple of the diff base using the the list of filenames to be
```
Comment (dhermes) [inline scripts/script_utils.py]: Just noticed "the the" in this docstring. Take this opportunity to fix it? (Almost certainly I am the owner of the original typo.)

**Pull request merged**